support/oidc-discovery-provider/README.md

```diff
@@ -0,0 +1,103 @@
+# SPIRE OIDC Discovery Provider
+
+The SPIRE OIDC Discovery Provider is a small helper that provides a minimal
+implementation of a subset of the OIDC discovery document as related to
+exposing a JSON Web Key Set (JWKS) for JSON Web Token (JWT) validation.
+
+It provides the following endpoints:
+
+| Verb  | Path                                | Description                               |
+| ----- | ------------------------------------| ------------------------------------------|
+| `GET` | `/.well-known/openid-configuration` | Returns the OIDC configuration document   |
+| `GET` | `/keys`                             | Returns the JWKS for JWT validation       |
+
+The helper by default relies on ACME to obtain TLS certificates that it uses to
+serve the documents securely.
+
+## Configuration
+
+### Command Line Configuration
+
+The registrar has the following command line flags:
+
+| Flag         | Description                                                      | Default                       |
+| ------------ | -----------------------------------------------------------------| ----------------------------- |
+| `-config`    | Path on disk to the [HCL Configuration](#hcl-configuration) file | `oidc-discovery-provider.conf` |
+
+
+### HCL Configuration
+
+The configuration file is a **required** by the registrar. It contains
+[HCL](https://github.com/hashicorp/hcl) encoded configurables.
+
+| Key                | Type    | Required? | Description                              | Default |
```

Comment:
nit: would be nice if this list was ordered by required/optional and/or alphabetical

Reply:
👍